operators/multiclusterobservability/controllers/placementrule/placementrule_controller.go

```diff
@@ -591,7 +591,8 @@ func areManagedClusterLabelsReady(obj client.Object) bool {
 	if _, openshiftVersionPresent := obj.GetLabels()["openshiftVersion"]; openshiftVersionPresent && openshiftVendor {
 		return true
 	}
-	if vendorOk && vendor != "auto-detect" {
+
+	if vendorOk && vendor != "auto-detect" && vendor != "OpenShift" {
```

Comment:
Could we document what this function has to do in a comment/doc above it? It's written in a way that's very difficult to understand. I also suggest a small refactor: 

```go
func areManagedClusterLabelsReady(obj client.Object) bool {
	vendor, foundVendor := obj.GetLabels()["vendor"]
	if !foundVendor || vendor == "" {
		log.Info("ManagedCluster labels are not ready", "cluster", obj.GetName())
		return false
	}

	if vendor == "OpenShift" {
		_, foundOpenshiftVersion := obj.GetLabels()["openshiftVersion"]
		return foundOpenshiftVersion
	}

	if foundVendor && vendor != "auto-detect" {
		return true
	}


	log.Info("ManagedCluster labels are not ready", "cluster", obj.GetName())
	return false
}
```

Please double check that the logic I wrote in the function comment is correct.

Reply:
Hum, I ate a log line when there's no `openshiftVersion` label. Also could simplify even more:

```go
// areManagedClusterLabelsReady check if labels automatically set in the managed cluster
// are ready to be accessed. These labels are: "vendor" and "openshiftVersion".
// Labels are considered not ready when:
//
// - The "vendor" label isn't found.
// - The "vendor" label has value "auto-detect".
// - The "vendor" has value "openshift" but there is no "openshiftVersion".
//
func areManagedClusterLabelsReady(obj client.Object) bool {
	labels := obj.GetLabels()
	vendor, foundVendor := labels["vendor"]

	if !foundVendor || vendor == "" || vendor == "auto-detect" {
		log.Info("ManagedCluster labels are not ready", "cluster", obj.GetName())
		return false
	}

	if vendor == "OpenShift" {
		_, foundOpenshiftVersion := labels["openshiftVersion"]
		if !foundOpenshiftVersion {
			log.Info("ManagedCluster labels are not ready", "cluster", obj.GetName())
                        return false
		}
	}

	return true
}
```